lively.morphic/world.js

```diff
@@ -332,8 +332,14 @@ export class World extends Morph {
 
   visibleBounds () {
     // the bounds call seems to slow down halos...
-    if (!this.env.renderer) return this.innerBounds();
-    return this.windowBounds().intersection(this.innerBounds());
+    let visibleBounds;
+    if (!this.env.renderer) visibleBounds = this.innerBounds();
+    else visibleBounds = this.windowBounds().intersection(this.innerBounds());
+    // if not the whole world is visible, we have to adjust for scrollbars
+    // it does not matter in which direction the space is too small, since we always get both scroll bars
+    if (visibleBounds.height < this.extent.y || visibleBounds.width < this.extent.x) visibleBounds.width = visibleBounds.width - this.scrollbarOffset.x;
```

Comment:
This is only a problem on windows and linux. MacOS makes the scroll bar transparent by default. Consider checking for the OS here.